src/modes/draw_polygon.js

```diff
@@ -41,7 +37,17 @@ module.exports = function(ctx) {
       doubleClickZoom.disable(ctx);
       ctx.ui.queueMapClasses({ mouse: Constants.cursors.ADD });
       ctx.ui.setActiveButton(Constants.types.POLYGON);
-      this.on('mousemove', CommonSelectors.true, handleMouseMove);
+      this.on('mousemove', CommonSelectors.isVertex, () => {
+        ctx.ui.queueMapClasses({ mouse: Constants.cursors.POINTER });
+      });
+      this.on('click', CommonSelectors.isVertex, () => {
+        polygon.removeCoordinate(`0.${currentVertexPosition}`);
+        currentVertexPosition--;
```

Comment:
Yea... I'll look into this more. My first pass didn't have it but than there was an extra point.